src/resolve/resolved-dns-stub.c

```
@@ -431,6 +456,64 @@ static int manager_dns_stub_udp_fd(Manager *m) {
+
+        if (l->fd >= 0)
+                return 0;
+
+        fd = socket(socket_address_family(&l->address), SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0) {
+                r = -errno;
+                goto fail;
+        }
+
+        r = setsockopt_int(fd, IPPROTO_IP, IP_FREEBIND, true);
+        if (r < 0)
+                goto fail;
+
+        r = set_dns_stub_common_socket_options(fd);
+        if (r < 0)
+                goto fail;
+
+        if (bind(fd, &l->address.sockaddr.sa, l->address.size) < 0) {
+                r = -errno;
+                goto fail;
+        }
+
+        r = sd_event_add_io(m->event, &l->dns_stub_extra_event_source, fd, EPOLLIN, on_dns_stub_packet, m);
+        if (r < 0)
+                goto fail;
+
+        (void) sd_event_source_set_description(l->dns_stub_extra_event_source, "dns-stub-udp-extra");
+
+        l->fd = TAKE_FD(fd);
+
+        if (DEBUG_LOGGING) {
+                (void) sockaddr_pretty(&l->address.sockaddr.sa, FAMILY_ADDRESS_SIZE(l->address.sockaddr.sa.sa_family), true, true, &pretty);
+                log_debug("Listening on UDP socket %s.", strnull(pretty));
+        }
+
+        return 0;
+
+ fail:
+       (void) sockaddr_pretty(&l->address.sockaddr.sa, FAMILY_ADDRESS_SIZE(l->address.sockaddr.sa.sa_family), true, true, &pretty);
+       if (r == -EADDRINUSE)
+               return log_warning_errno(r,
+                                        "Another process is already listening on UDP socket %s.\n"
+                                        "Turning off local DNS stub extra support.", strnull(pretty));
+       else if (r == -EPERM)
+               return log_warning_errno(r,
+                                        "Failed to listen on UDP socket %s: %m.\n"
+                                        "Turning off local DNS stub extra support.", strnull(pretty));
+
+       assert(r < 0);
+
+       return log_error_errno(r, "Failed to listen on UDP socket %s: %m", strnull(pretty));
```

Those messages say that the failure is fatal: "Turning off ...", "Failed ...". In fact, failure is ignored. Please adjust the messages to say, ', ignoring', and downgrade this message to warning too.